src/app/InteractionModelEngine.cpp

```diff
@@ -220,6 +220,22 @@ CHIP_ERROR InteractionModelEngine::ShutdownSubscription(uint64_t aSubscriptionId
     return err;
 }
 
+CHIP_ERROR InteractionModelEngine::ShutdownSubscriptionsByNodeId(NodeId aPeerNodeId)
+{
+    CHIP_ERROR err = CHIP_ERROR_KEY_NOT_FOUND;
+
+    for (ReadClient & readClient : mReadClients)
+    {
+        if (!readClient.IsFree() && readClient.IsSubscriptionType() && readClient.GetPeerNodeId() == aPeerNodeId)
```

Comment:
There need to be checks for fabric here too, no?

Reply:
Hmm... `ReadClient` currently does not have fabric information. I guess I'll need to add it.

Reply:
Done.